internal/controller/etcdcluster_controller.go

```diff
@@ -100,6 +118,17 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
+	// Create Server and Peer Certificate for etcd-operator to communicate within the members
+	if etcdCluster.Spec.TLS != nil {
+		createServerPeerCertErr := r.checkServerPeerCertificate(etcdCluster, sts, ctx)
+		if createServerPeerCertErr != nil {
+			logger.Error(createServerPeerCertErr, "Error creating Server or Peer Certificate")
+		}
+	} else {
+		// TODO: instead of logging error, set default autoConfig
+		logger.Error(fmt.Errorf("missing TLS config for %s", etcdCluster.Name), "certificates cannot be created")
+	}
+
```

Comment:
Add a bool parameter for `reconcileStatefulSet`
- true: add member
- false: remove member

Get all the certificate management encapsulated in the function `reconcileStatefulSet`
- It doesn’t populate the reconciliation workflow.
- It’s easy to maintain the certificate management in one place instead of multiple places.